plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -1178,3 +1288,235 @@ SELECT
 	,s.[total_scheduler_delay_ms]
 FROM sys.dm_os_schedulers AS s
 `
+
+const sqlAzureMIPartQueryPeriod = `
+DECLARE @QueryData NVARCHAR(MAX) = ?1; --input parameter
+
+IF SERVERPROPERTY('EngineEdition') != 8
+	RETURN;
+
+IF OBJECT_ID ('tempdb.dbo.#ExecForeachDb') IS NOT NULL DROP PROCEDURE #ExecForeachDb;
+EXEC('CREATE PROCEDURE #ExecForeachDb @queryText NVARCHAR(MAX)
+AS
+BEGIN
+	SET NOCOUNT ON;
+
+	DECLARE dbCursor CURSOR LOCAL FOR 
+	SELECT 
+		[name] 
+	FROM sys.databases
+	WHERE HAS_DBACCESS([name]) = 1 
+		AND [name] NOT IN (''master'', ''tempdb'', ''model'', ''msdb'');
+	DECLARE @CurrSqlText nvarchar(max);
+	DECLARE @dbName sysname;
+
+	OPEN dbCursor;
+	FETCH NEXT FROM dbCursor INTO @dbName;
+
+	WHILE @@Fetch_Status=0 BEGIN
+		SET @CurrSqlText = N''USE ''+ QUOTENAME(@dbName ,''"'') + @queryText
+		EXEC sp_executesql @CurrSqlText
+		FETCH NEXT FROM dbCursor INTO @dbName;
+	END
+
+	CLOSE dbCursor;
+	DEALLOCATE dbCursor;
+END');
+
+IF OBJECT_ID ('tempdb.dbo.#CachedDbTimestamps') IS NOT NULL DROP TABLE #CachedDbTimestamps;
+CREATE TABLE #CachedDbTimestamps (
+	[id] INT,
+	[timestamp] BIGINT
+);
+
+DECLARE @XmlQueryData XML = @QueryData;
+INSERT INTO #CachedDbTimestamps
+SELECT
+	s.value('./id[1]', 'int') AS [id],
+	s.value('./timestamp[1]', 'bigint') AS [timestamp]
+FROM @XmlQueryData.nodes('/Timestamps/ts') t(s);
+
+IF OBJECT_ID ('tempdb.dbo.#DbTimestamps') IS NOT NULL DROP TABLE #DbTimestamps;
+CREATE TABLE #DbTimestamps (
+	[id] INT,
+	[timestamp] BIGINT
+);
+
+DECLARE @SqlText NVARCHAR(MAX) = N'
+
+DECLARE @lastIntervalEndTimestamp BIGINT = 0;
+SELECT TOP 1 @lastIntervalEndTimestamp = [timestamp] FROM #CachedDbTimestamps WHERE [id] = DB_ID();
+
+INSERT INTO #DbTimestamps([id], [timestamp]) VALUES(DB_ID(), @lastIntervalEndTimestamp)
+
+DECLARE @magicNumberDays BIGINT = 4294967296, @magicNumberSeconds BIGINT = 300, @secondsInDay INT = 86400;
+
+DECLARE @timeGrain BIGINT = (SELECT interval_length_minutes * 60 FROM sys.database_query_store_options WITH (NOLOCK)); --use query store time grain
+
+DECLARE @currDate DATETIME = GETUTCDATE();
```

Comment:
Please add comments of what these time calculations are doing for maintainability.

Reply:
Added comments